components/app-proxy/vm.tf

```diff
@@ -6,78 +6,65 @@ module "tags" {
   builtFrom   = var.builtFrom
 }
 
-resource "azurerm_resource_group" "this" {
+resource "azurerm_resource_group" "app_proxy_rg" {
   name     = "${var.product}-${var.env}-rg"
   location = var.location
 
   tags = module.tags.common_tags
 }
 
-# TODO store in a vault
-resource "random_password" "this" {
-  length = 30
-}
-
-resource "azurerm_subnet" "this" {
-  name = "app-proxy"
-
+resource "azurerm_subnet" "app_proxy_subnet" {
+  name                 = "app-proxy"
   address_prefixes     = ["10.99.73.0/25"]
-  resource_group_name  = "mgmt-vpn-2-mgmt"
-  virtual_network_name = "mgmt-vpn-2-vnet"
+  resource_group_name  = var.app_proxy_subnet_rg
+  virtual_network_name = var.app_proxy_vnet_name
 }
 
 # TODO auto-register VMs in DNS with private dns autoregistration
 module "virtual_machine" {
-  source = "git::https://github.com/hmcts/terraform-module-virtual-machine.git?ref=master"
-
-  vm_type              = "windows"
-  vm_name              = "${var.product}-${var.env}"
-  vm_resource_group    = azurerm_resource_group.this.name
-  vm_admin_password    = random_password.this.result
-  vm_subnet_id         = azurerm_subnet.this.id
-  vm_publisher_name    = "MicrosoftWindowsServer"
-  vm_offer             = "WindowsServer"
-  vm_sku               = "2022-datacenter-azure-edition-core"
-  vm_size              = "Standard_D2ds_v5"
-  vm_version           = "latest"
-  vm_availabilty_zones = "1"
```

Comment:
I don't think there's any reason to extract these to variables currently, it's just a layer of indirection. we can add as required

Reply:
Sure thing, added one or two in case I needed to switch them and the rest followed - will keep as it was